irohad/subscription/scheduler_impl.hpp

```diff
@@ -0,0 +1,163 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef IROHA_SUBSCRIPTION_SCHEDULER_IMPL_HPP
+#define IROHA_SUBSCRIPTION_SCHEDULER_IMPL_HPP
+
+#include <assert.h>
+#include <algorithm>
+#include <chrono>
+#include <deque>
+#include <functional>
+#include <mutex>
+#include <shared_mutex>
+#include <thread>
+
+#include "subscription/scheduler.hpp"
+
+#include "subscription/common.hpp"
+
+/**
+ * If you need to execute task, that was made in this thread and want to be
+ * executed in the same thread without delay - you need to uncomment this define
+ */
+//#define SE_SYNC_CALL_IF_SAME_THREAD
+
+namespace iroha::subscription {
+
+  class SchedulerBase : public IScheduler, utils::NoCopy, utils::NoMove {
+   private:
+    using Time = std::chrono::high_resolution_clock;
```

Comment:
I'd prefer same terminology as std:: and boost::
```suggestion
    using Clock = std::chrono::high_resolution_clock;
```

Reply:
I like ```time``` and ```timestamp``` much more than ```clock``` and ```time-point```.